src/kubernetes.cc

```diff
@@ -796,6 +796,14 @@ void KubernetesReader::WatchMaster(
   if (verbose) {
     LOG(INFO) << "WatchMaster(" << name << "): Contacting " << endpoint;
   }
+  const int timeout = config_.HealthCheckWatchTimeoutSeconds();
+  std::mutex expiration_mutex;
+  auto expiration = std::chrono::steady_clock::now() + time::seconds(timeout);
```

Comment:
I would add a comment here explaining what this is, something like "The time by when the watcher has to receive some data to be considered healthy"...